### PR TITLE
Read and write json files

### PR DESCRIPTION
**Pull request by iskandr**
Added methods to `VariantCollection`:
* `variants.to_json_file(filename)`
* `VariantCollection.from_json_file(filename)`

Also fixed initialization of `end` property from basic fields.

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/varcode/82)

### CODE REVIEW

File: varcode/variant.py [hunk 1]
```diff
@@ -223,9 +223,13 @@ def __setstate__(self, fields):
         self.ensembl = cached_release(fields.pop("release"))
         self.info = fields.pop("info", {})
 
-        # Remaining fields are simple properties that just get set.
+        # Remaining fields  are simple properties that just get set.
         self.__dict__.update(fields)
 
+        # TODO: switch to interbase coordinates so that 'end' becomes a
+        # mandatory rather than derived field
+        self.end = self.start + max(0, len(self.ref) - 1)
```
Comment (timodonnell) [inline varcode/variant.py]: This line may be right but it gives me pause. Can we add `end` to `fields()` instead? Then we would just extract it from the saved state here.

File: varcode/variant_collection.py [hunk 2]
```diff
@@ -191,10 +191,18 @@ def to_json(self):
         """
         return json.dumps(self.__getstate__())
 
+    def to_json_file(self, filename):
```
Comment (timodonnell) [inline varcode/variant_collection.py]: Nit: I think `write_json` might be a better name for this function

Comment (timodonnell) [inline varcode/variant_collection.py]: or `write_json_file`, then the reverse can be called `read_json_file`

File: varcode/variant.py (hunk 1 above)
Comment (iskandr) [inline varcode/variant.py]: Seems reasonable, but one problem is that the `__init__` method doesn't take an `end` argument. Maybe I should make `end` optional and only compute it from the trimmed ref if it's absent?

**Comment by timodonnell** [comment]
LGTM

**Pull request merged**